### PR TITLE
Add rosdep keys for ROS2 dependencies.

### PR DESCRIPTION
**Pull request by allenh1**
This adds keys for Gentoo.

@nuclearsandwich FYI

### CODE REVIEW

File: rosdep/python.yaml [hunk 1]
```diff
@@ -3200,25 +3201,32 @@ python3-babeltrace:
...
 python3-pep8:
+  gentoo: [dev-python/pep8]
   ubuntu:
     xenial: [python3-pep8]
 python3-pyparsing:
+  gentoo: [dev-python/pyparsing]
   ubuntu:
     xenial: [python3-pyparsing]
 python3-nose:
+  gentoo: [dev-python/setuptools]
```
Comment (mikaelarguedas) [inline rosdep/python.yaml]: typo / copy-n-paste?

Comment (allenh1) [reply rosdep/python.yaml]: typo/brain-malfunction, actually (and sadly). Thanks.

**Comment by mikaelarguedas** [comment]
@nuclearsandwich to confirm, but I think that we use the rosdep database from ros/rosdistro for ROS 2 as well. So this PR should be open on the upstream repo if that's the case

**Comment by allenh1** [comment]
> So this PR should be open on the upstream repo if that's the case

Regardless of where the rosdep database pulls from, this should probably still be filed on the upstream repo. If preferred, I can do that and attempt to rebase, but I'm terrified of trying that. @mikaelarguedas should I submit this to both repos?

**Comment by dirk-thomas** [comment]
It would be best if the `rosdep` directory in this repo would be deleted since it is not used.

**Comment by mikaelarguedas** [comment]
>  If preferred, I can do that and attempt to rebase, but I'm terrified of trying that. @mikaelarguedas should I submit this to both repos?

Let's get this approved in one of the repos then we can just apply the commit to the other repo. Without having to rebase

**Comment by mikaelarguedas** [comment]
> It would be best if the `rosdep` directory in this repo would be deleted since it is not used.

Agreed, and if it's currently used we should look into consolidating the 2 databases as submitting PRs to both repos is not very convenient moving forward

**Comment by allenh1** [comment]
@dirk-thomas @mikaelarguedas removing this directory sounds quite preferential to me. I noticed that the keys exist in both repositories, but the ros/rosdistro keys apply to ubuntu and debian where as the ros2/rosdistro apply only to ubuntu.

I'll go ahead and file the PR to ros/rosdistro and close this one.

Pending input from @nuclearsandwich, I'll file a PR into this repository that removes the `rosdep` directory, if it is not needed.

**Comment by mikaelarguedas** [comment]
> Pending input from @nuclearsandwich, I'll file a PR into this repository that removes the rosdep directory, if it is not needed.

let's wait for feedback before opening PR that may not make sense as we lack the relevant information as of what is used by the ros2 farm at the moment.

**Comment by allenh1** [comment]
> let's wait for feedback before opening PR that may not make sense as we lack the relevant information as of what is used by the ros2 farm at the moment.

Well, My bad. I failed to see this and filed the PR. Sorry for the noise.

**Comment by mikaelarguedas** [comment]
That's fine I was talking about the PR removing the rosdep directory ;) It's good that you opened the PR adding the rules upstream regardless of what db the ROS 2 infrastructure uses :+1:

**Comment by dirk-thomas** [comment]
> let's wait for feedback before opening PR that may not make sense as we lack the relevant information as of what is used by the ros2 farm at the moment.

As I already commented before: `rosdep` uses the `ros/rosdistro` database. That can easily be confirmed by looking at the release instructions: https://github.com/ros2/ros2/wiki/Releasing-a-ROS-2-package-with-bloom Since they don't mention a custom `rosdep` URL `bloom` will use the default database.

**Comment by dirk-thomas** [comment]
@allenh1 Can the branch of this PR be removed?

**Comment by mikaelarguedas** [comment]
> Can the branch of this PR be removed?

I was waiting to know if we want the rosdep db here to be updated as well or not before deleting

**Comment by allenh1** [comment]
I was also waiting on the feedback, but I can certainly remove it if you need me to.

**Comment by nuclearsandwich** [comment]
During the beta2 bootstrap phase I was using this rosdep database for rapid development but the necessary keys have been pushed upstream this fork is now well out of sync.

I agree we should drop the rosdep keys from this fork. I'll PR that forthwith.

@allenh1 you can delete the branch attached to this PR at your leisure. I won't do it for you on the off chance you want to archive or pull it somewhere else first.

**Pull request closed without merging**